### ISSUE TITLE
Bug: skip_dir = "/crypt" does not work

### ISSUE THREAD

**Issue by kailuaken22** [abraunegg/onedrive#2590]
### Describe the bug

I have a top level directory called "crypt" (/crypt).  skip_dir does not work with that directory (it works with all other directories).  In other words, onedrive tries to backup all of the contents of the /crypt directory, even with the skip_dir directive in the config file.   (Manjaro Linux - 6.6.8-2-MANJARO #1 SMP PREEMPT_DYNAMIC Thu Dec 21 16:21:45 UTC 2023 x86_64 GNU/Linux, onedrive v2.4.25).  

### Operating System Details

```shell
uname -a
Linux laptop 6.6.8-2-MANJARO #1 SMP PREEMPT_DYNAMIC Thu Dec 21 16:21:45 UTC 2023 x86_64 GNU/Linux

lsb_release -a 
LSB Version:	n/a
Distributor ID:	ManjaroLinux
Description:	Manjaro Linux
Release:	23.1.2
Codename:	Vulcan

onedrive --version
onedrive v2.4.25
```

### Client Installation Method

From 3rd Party Source (PPA, OpenSuSE Build Service etc)

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

v2.4.25

### What is your OneDrive Application Configuration

```shell
onedrive --display-config
Configuration file successfully loaded
onedrive version                             = v2.4.25
Config path                                  = /home/kjc/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /home/kjc/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = /Desktop|/Downloads|/Email attachments|/lynmarie-hula-videos|/Music|/NAVRES|/Personal Vault|/Pictures|/Public|/Crypt|/crypt|/Documents|/MITRE|/crypt/
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.bin
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 600
Config option 'ip_protocol_version'          = 0
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

### What is your 'curl' version

```shell
curl --version                                                   ✔ 
curl 8.5.0 (x86_64-pc-linux-gnu) libcurl/8.5.0 OpenSSL/3.2.0 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.4 libpsl/0.21.2 (+libidn2/2.3.4) libssh2/1.11.0 nghttp2/1.58.0
Release-Date: 2023-12-06
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
mount                                                            ✔ 
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
sys on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
dev on /dev type devtmpfs (rw,nosuid,relatime,size=1936952k,nr_inodes=484238,mode=755,inode64)
run on /run type tmpfs (rw,nosuid,nodev,relatime,mode=755,inode64)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on / type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=256,subvol=/@)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,inode64)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=34,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=4368)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,nosuid,nodev,relatime,pagesize=2M)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /tmp type tmpfs (rw,noatime,inode64)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /var/cache type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=258,subvol=/@cache)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /home type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@home)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /var/log type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=259,subvol=/@log)
/dev/nvme0n1p1 on /boot/efi type vfat (rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=391384k,nr_inodes=97846,mode=700,uid=1000,gid=1000,inode64)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
```

### What are all your local file system partition types

```shell
mount                                                            ✔ 
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
sys on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
dev on /dev type devtmpfs (rw,nosuid,relatime,size=1936952k,nr_inodes=484238,mode=755,inode64)
run on /run type tmpfs (rw,nosuid,nodev,relatime,mode=755,inode64)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on / type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=256,subvol=/@)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,inode64)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=34,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=4368)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,nosuid,nodev,relatime,pagesize=2M)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /tmp type tmpfs (rw,noatime,inode64)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /var/cache type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=258,subvol=/@cache)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /home type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=257,subvol=/@home)
/dev/mapper/luks-4154f3b0-d243-4593-b31e-cf9cf0ae6998 on /var/log type btrfs (rw,relatime,ssd,discard=async,space_cache=v2,subvolid=259,subvol=/@log)
/dev/nvme0n1p1 on /boot/efi type vfat (rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=ascii,shortname=mixed,utf8,errors=remount-ro)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=391384k,nr_inodes=97846,mode=700,uid=1000,gid=1000,inode64)
gvfsd-fuse on /run/user/1000/gvfs type fuse.gvfsd-fuse (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
httpdirfs on /home/kjc/.cache/yay/ttf-ms-win11-auto/src/mnt/http type fuse.httpdirfs (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
```

### How do you use 'onedrive'

OneDrive account is shared across multiple systems (1 Windows 11 system and 1 Fedora system).  

### Steps to reproduce the behaviour

 onedrive --synchronize --verbose --resync --dry-run shows that the /crypt directory is getting backed up.

### Complete Verbose Log Output

```shell
Using 'user' Config Dir: /home/kjc/.config/onedrive
Using 'system' Config Dir: /etc/onedrive
Configuration file successfully loaded

The use of --resync will remove your local 'onedrive' client state, thus no record will exist regarding your current 'sync status'
This has the potential to overwrite local versions of files with potentially older versions downloaded from OneDrive which can lead to data loss
If in-doubt, backup your local data first before proceeding with --resync

Are you sure you wish to proceed with --resync? [Y/N] 
config file has been updated, checking if --resync needed
DRY-RUN Configured. Output below shows what 'would' have occurred.
Deleting the saved application sync status ...
Checking Application Version ...
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Using Curl defaults for all HTTP operations
Opening the item database ...
All operations will be performed in: /home/kjc/OneDrive
Application version: v2.4.25
Account Type: personal
Default Drive ID: 608fa005d147c64a
Default Root ID: 608FA005D147C64A!128
Remaining Free Space: 703623017267
Fetching details for OneDrive Root
OneDrive Root does not exist in the database. We need to add it.
Added OneDrive Root to the local database
Initializing the Synchronization Engine ...
Syncing changes and items from OneDrive ...
Applying changes of Path ID: 608FA005D147C64A!128
Updated Remaining Free Space: 703623017267
Processing 213 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Documents
Creating local directory: ./crypt
Skipping item - file path is excluded by skip_dir config: crypt/test_file.txt
Creating local directory: crypt/bacula
Creating local directory: crypt/bacula-test
Downloading file crypt/bacula/test_file.txt ... done.
Downloading file crypt/bacula-test/test_file.txt ... done.
Processing 208 OneDrive items to ensure consistent local state due to a full scan being triggered by actions on OneDrive
Skipping item - excluded by skip_dir config: /Documents
Processing 205 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Documents
Processing 211 OneDrive items to ensure consistent local state
...
Skipping item - excluded by skip_dir config: /Documents
Processing 219 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Desktop
Skipping item - excluded by skip_dir config: /Downloads
Skipping item - excluded by skip_dir config: /Pictures
Skipping item - excluded by skip_dir config: /Documents
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9723
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!490
Creating local directory: crypt/bacula-test/storage
Creating local directory: crypt/bacula-test/working
Creating local directory: crypt/bacula-test/offsite-backup-storage
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!13845
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!13850
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!13844
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!13851
Skipping item - excluded by skip_file config: OffsiteFullCatalog-0647.bin
Skipping item - excluded by skip_file config: OffsiteFullCatalog-0611.bin
Skipping item - excluded by skip_file config: OffsiteFullCatalog-0644.bin
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!13895
Skipping item - excluded by skip_file config: OffsiteFullDisk-0654.bin
Skipping item - excluded by skip_file config: OffsiteFullDisk-0655.bin
Skipping item - excluded by skip_file config: OffsiteFullDisk-0653.bin
Skipping item - excluded by skip_file config: OffsiteFullDisk-0652.bin
Skipping item - excluded by skip_file config: OffsiteFullDisk-0659.bin
Skipping item - excluded by skip_file config: OffsiteFullDisk-0658.bin
...
Skipping item - excluded by skip_dir config: /Music
Skipping item - excluded by skip_dir config: /Desktop
Skipping item - excluded by skip_dir config: /Downloads
Skipping item - excluded by skip_dir config: /Pictures
Skipping item - excluded by skip_dir config: /Documents
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9723
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!490
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9725
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!12633
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9718
Skipping item - excluded by skip_file config: Vol-0014.bin
Skipping item - excluded by skip_file config: Vol-0013.bin
Skipping item - excluded by skip_file config: Vol-0012.bin
Skipping item - excluded by skip_file config: Vol-0015.bin
Skipping item - excluded by skip_file config: Vol-0016.bin
Skipping item - excluded by skip_file config: Vol-0019.bin
Skipping item - excluded by skip_file config: Vol-0017.bin
Skipping item - excluded by skip_file config: Vol-0018.bin
Skipping item - excluded by skip_file config: Vol-0021.bin
Skipping item - excluded by skip_file config: Vol-0025.bin
Skipping item - excluded by skip_file config: Vol-0026.bin
Skipping item - excluded by skip_file config: Vol-0024.bin
Skipping item - excluded by skip_file config: Vol-0027.bin
Skipping item - excluded by skip_file config: Vol-0023.bin
Skipping item - excluded by skip_file config: Vol-0020.bin
Skipping item - excluded by skip_file config: Vol-0022.bin
Skipping item - excluded by skip_file config: Vol-0002.bin
Skipping item - excluded by skip_file config: Vol-0005.bin
Skipping item - excluded by skip_file config: Vol-0003.bin
Skipping item - excluded by skip_file config: Vol-0004.bin
Skipping item - excluded by skip_file config: Vol-0006.bin
Skipping item - excluded by skip_file config: Vol-0007.bin
Skipping item - excluded by skip_file config: Vol-0008.bin
Skipping item - excluded by skip_file config: Vol-0009.bin
Skipping item - excluded by skip_file config: Vol-0010.bin
Skipping item - excluded by skip_file config: Vol-0011.bin
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!10067
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9909
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!9724
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!7824
Internet connectivity to Microsoft OneDrive service has been restored

  Calling Function: applyDifferences()

Performing a database consistency and integrity check on locally stored data ... 
Uploading differences of ~/OneDrive
Processing .
The directory has not changed
...
Processing 203 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Pictures
Processing 206 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Pictures
Skipping item - excluded by skip_dir config: /Documents
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!490
Processing 206 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Pictures
Skipping item - excluded by skip_dir config: /Documents
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!490
Processing 33 OneDrive items to ensure consistent local state
Skipping item - excluded by skip_dir config: /Email attachments
Skipping item - excluded by skip_dir config: /Music
Skipping item - excluded by skip_dir config: /Public
Skipping item - excluded by skip_dir config: /Downloads
Skipping item - excluded by skip_dir config: /Pictures
Skipping item - excluded by skip_dir config: /Documents
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!490
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!493
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!497
WARNING: OneDrive API inconsistency - this file does not have any hash: 608FA005D147C64A!496
Number of items from OneDrive to process: 1
Sync with OneDrive is complete
```

**Comment by abraunegg** [comment]
@kailuaken22 
> I have a top level directory called "crypt" (/crypt). skip_dir does not work with that directory (it works with all other directories). In other words, onedrive tries to backup all of the contents of the /crypt directory, even with the skip_dir directive in the config file

Thankyou for raising this curious issue. I do note however you have listed 'crypt' 3 different times - I am guessing to try and work out potential folder name issues:

* `/Crypt`
* `/crypt`
* `/crypt/`

Using your *exact* `skip_dir` configuration, and with the following *online* folders:
#### Top Level
![image](https://github.com/abraunegg/onedrive/assets/4956234/32767605-89cf-4412-aeb2-91bb29980376)
 
#### 1st level
![image](https://github.com/abraunegg/onedrive/assets/4956234/4a7f0938-b4c4-424d-9633-fddd684fa490)

#### 2nd Level
![image](https://github.com/abraunegg/onedrive/assets/4956234/b941a904-fb5a-4d4b-b6e5-4b1d8937aa65)

#### Application Configuration:
```
[alex@onedrive-client-dev onedrive]$ ./onedrive --display-config
Configuration file successfully loaded

Unable to access /var/log/onedrive/
Please manually create '/var/log/onedrive/' and set appropriate permissions to allow write access
The requested client activity log will instead be located in your users home directory
onedrive version                             = v2.4.25-13-g1a88d33
Config path                                  = /home/alex/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /home/alex/OneDrive
Config option 'enable_logging'               = true
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     = /Desktop|/Downloads|/Email attachments|/lynmarie-hula-videos|/Music|/NAVRES|/Personal Vault|/Pictures|/Public|/Crypt|/crypt|/Documents|/MITRE|/crypt/
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = true
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 0
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = true
Config option 'check_nomount'                = true
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = 
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   = 
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 600
Config option 'ip_protocol_version'          = 1
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = true
sync_list contents:
# Comments
# Exclude random_25k_files
!/random_25k_files/*

# Include everything else
/*

```

I can confirm that this *odd* bug exists for in v2.4.x codebase:

```
Checking Application Version ...
Initializing the OneDrive API ...
Configuring Global Azure AD Endpoints
Using Curl defaults for all HTTP operations
Opening the item database ...
All operations will be performed in: /home/alex/OneDrive
Application version: v2.4.25-13-g1a88d33
Account Type: personal
Default Drive ID: 66d53be8a5056eca
Default Root ID: 66D53BE8A5056ECA!101
Remaining Free Space: 5271718518
Fetching details for OneDrive Root
OneDrive Root does not exist in the database. We need to add it.
Added OneDrive Root to the local database
Initializing the Synchronization Engine ...
Syncing changes and items from OneDrive ...
Applying changes of Path ID: 66D53BE8A5056ECA!101
Updated Remaining Free Space: 5271718518
Processing 5 OneDrive items to ensure consistent local state due to sync_list being used
Creating local directory: ./crypt
Creating local directory: crypt/bacula
Downloading file crypt/bacula/test_file.txt.txt ... done.
Skipping item - .file or .folder: ./
Processing 1 OneDrive items to ensure consistent local state due to a full scan being triggered by actions on OneDrive
Performing a database consistency and integrity check on locally stored data ... 
Uploading differences of ~/OneDrive
Processing .
The directory has not changed
Processing crypt
The directory has not changed
Uploading new items of ~/OneDrive
Skipping item - excluded by sync_list config: ./random_25k_files
Skipping item - .file or .folder: ./.folder
Applying changes of Path ID: 66D53BE8A5056ECA!101
Updated Remaining Free Space: 5271718518
Processing 1 OneDrive items to ensure consistent local state due to a full scan being requested
Sync with OneDrive is complete
```

Unfortunately, the issue here this is *not* going to be fixed in v2.4.x, however, whilst the code logic for v2.5.x was similar, this issue has been fixed and is not present in v2.5.0 'alpha-5':

#### Application Configuration
```
[alex@onedrive-client-dev onedrive-pr2584]$ ./onedrive --display-config
Reading configuration file: /home/alex/.config/onedrive/config
Configuration file successfully loaded

ERROR: Unable to create /var/log/onedrive
ERROR: Please manually create '/var/log/onedrive' and set appropriate permissions to allow write access for your user to this location.
ERROR: The requested client activity log will instead be located in your users home directory

onedrive version                             = v2.5.0-alpha-5 GitHub version: v2.4.25-28-g0ce088f
Config path                                  = /home/alex/.config/onedrive
Config file found in config path             = true
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = ~/OneDrive
Config option 'enable_logging'               = true
Config option 'log_dir'                      = /var/log/onedrive
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     = /Desktop|/Downloads|/Email attachments|/lynmarie-hula-videos|/Music|/NAVRES|/Personal Vault|/Pictures|/Public|/Crypt|/crypt|/Documents|/MITRE|/crypt/
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = true
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 0
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = true
Config option 'check_nomount'                = true
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = 
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.0-alpha-5
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 30
Config option 'data_timeout'                 = 240
Config option 'ip_protocol_version'          = 1

Selective sync 'sync_list' configured        = true
sync_list config option 'sync_root_files'    = false
sync_list contents:
# Comments
# Exclude random_25k_files
!/random_25k_files/*

# Include everything else
/*

Config option 'sync_business_shared_items'   = false
Selective Business Shared Items configured   = false

Config option 'webhook_enabled'              = false
```

#### Application Run
```
[alex@onedrive-client-dev onedrive-pr2584]$ ./onedrive --synchronize --verbose --resync --resync-auth --dry-run 
Reading configuration file: /home/alex/.config/onedrive/config
Configuration file successfully loaded
Using 'user' configuration path for application state data: /home/alex/.config/onedrive

ERROR: Unable to create /var/log/onedrive
ERROR: Please manually create '/var/log/onedrive' and set appropriate permissions to allow write access for your user to this location.
ERROR: The requested client activity log will instead be located in your users home directory

Using the following path to store the runtime application log: /home/alex

DEPRECIATION WARNING: --synchronize has been depreciated in favour of --sync or -s
DEPRECIATION WARNING: Depreciated commands will be removed in a future release.

DRY-RUN Configured. Output below shows what 'would' have occurred.
DRY-RUN: No database copy created for --dry-run due to --resync also being used
DRY RUN: Not removing hash files as --dry-run has been used
DRY RUN: Not creating backup config file as --dry-run has been used
DRY RUN: Not updating hash files as --dry-run has been used
Deleting the saved application sync status ...
DRY RUN: Not removing the saved application sync status
Forcing client to use IPv4 connections only
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  v2.5.0-alpha-5 GitHub version: v2.4.25-28-g0ce088f
Account Type:         personal
Default Drive ID:     66d53be8a5056eca
Default Root ID:      66D53BE8A5056ECA!101
Remaining Free Space: 4.91 GB (5271718518 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/alex/OneDrive
Fetching /delta response from the OneDrive API for Drive ID: 66d53be8a5056eca
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 6
Skipping item - excluded by skip_dir config: /crypt
Skipping item - excluded by skip_dir config: /Music
Processing API Response Bundle: 2 - Quantity of 'changes|items' in this bundle to process: 1
Finished processing /delta JSON response from the OneDrive API
Processing 2 applicable changes and items received from Microsoft OneDrive 
Processing OneDrive JSON item batch [1/1] to ensure consistent local state
Performing a database consistency and integrity check on locally stored data ... 
Processing DB entries for this Drive ID: 66d53be8a5056eca
Processing ~/OneDrive
The directory has not changed
Scanning the local file system '~/OneDrive' for new data to upload ...
Skipping item - excluded by sync_list config: ./random_25k_files
Skipping item - .file or .folder: ./.folder
Performing a last examination of the most recent online data within Microsoft OneDrive to complete the reconciliation process
Fetching /delta response from the OneDrive API for Drive ID: 66d53be8a5056eca
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 0
Finished processing /delta JSON response from the OneDrive API
No additional changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive

Sync with Microsoft OneDrive is complete
[alex@onedrive-client-dev onedrive-pr2584]$ 
```

Given that no issue exists in `v2.5.0-alpha-5` I am marking this as fixed and closing this issue.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
